ddd/projet_doctoral/doctorat/formation/commands.py

```diff
@@ -0,0 +1,89 @@
+# ##############################################################################
+#
+#    OSIS stands for Open Student Information System. It's an application
+#    designed to manage the core business of higher education institutions,
+#    such as universities, faculties, institutes and professional schools.
+#    The core business involves the administration of students, teachers,
+#    courses, programs and so on.
+#
+#    Copyright (C) 2015-2022 Université catholique de Louvain (http://www.uclouvain.be)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    A copy of this license - GNU General Public License - is available
+#    at the root of the source code of this program.  If not,
+#    see http://www.gnu.org/licenses/.
+#
+# ##############################################################################
+from datetime import date
+from typing import List, Optional
+
+import attr
+
+from osis_common.ddd.interface import CommandRequest
+
+
+@attr.dataclass
+class AjouterActiviteCommand(CommandRequest):
+    doctorat_uuid: str
+    ects: float
+
+
+@attr.dataclass
+class ModifierActiviteCommand(CommandRequest):
+    activite_uuid: str
+    ects: float
```

Comment:
Le terme "Modifier" fait très "CRUD" ; quelle est l'opération métier exactement ? "Modifier les crédits de l'activité" ?

Reply:
D'après l'analyse métier, on est clairement sur du CRUD https://confluence.sgsi.ucl.ac.be/display/OSIS/Formation+doctorale#Formationdoctorale-CONFERENCES

J'ai essayé de faire "moins" de CRUD en intégrant la notion d'activité (qui comprend statut et ects) et donc d'avoir un seul use case pour encoder les détails d'une activité (que ce soit Publication/Communication/Conférence).

Mais du coup je sais pas si c'est judicieux ?

Reply:
Ouii c'est l'impression que j'ai eue en lisant la PR ; si la demande métier est vraiment de juste capturer de l'information sans complexité métier, on pourrait se passer du DDD et faire une simple saisie formulaire directe en DB. Maintenant, si la gestion de l'activité (mais pas la captures de toutes ces infos) est plus complexe, on pourrait aussi imaginer avoir juste l'activité et sa gesiton (workflow, etc) en DDD et capturer le reste en CRUD. À voir la solution la plus simple à mettre en place en fonction du besoin ?

Reply:
@basmot Bien que ta remarque soit pertinente, et étant donné que le développement est déjà réalisé, je pense que nous pouvons rester sur l'approche DDD. 
Pour info @SebCorbin , les termes "**CRUD**" doivent être au maximum éviter lors des phases d'analyse / conception, n'hésitez pas à challenger lors des analyses